crates/nu-command/src/filters/length.rs

```diff
@@ -81,7 +81,17 @@ fn length_row(call: &Call, input: PipelineData) -> Result<PipelineData, ShellErr
         PipelineData::Value(Value::Nothing { .. }, ..) => {
             Ok(Value::int(0, call.head).into_pipeline_data())
         }
-        _ => Ok(Value::int(input.into_iter().count() as i64, call.head).into_pipeline_data()),
+        _ => {
+            let mut count: i64 = 0;
+            // Check for and propagate errors
```

Comment:
I'm a little concerned by this approach of checking for errors in downstream commands.I worry that there will be a long tail of commands where we forget to do it, leading to inconsistency. It also takes us further away from an "errors are just data" approach (which we admittedly do not adhere to very well today).

That said, users who do something like `$foo | get bar | length` will expect it to fail on an unsuccessful `bar` access... so maybe this is an OK approach for now.

In the future it would be nice if this sort of check was handled at a higher level (not sure exactly where).